src/Monolog/Logger.php

```diff
@@ -311,6 +308,11 @@ public function addRecord(int $level, string $message, array $context = []): boo
             $record = call_user_func($processor, $record);
         }
 
+        reset($this->handlers);
+        while ($key !== key($this->handlers)) {
```

Comment:
@stloyd it actually does not matter - they are guaranteed to be the same value at this point